denovogear/denovogear
# Feature/inheritance model

Massive change to DeNovoGear internals to support Different models of inheritance.

Fixes #54, #216, #204, #157 
Updates Pull Requests #197, #196 
(and probably more)

## New Features
 - Support sex-linked inheritance models
 - Use slashes instead of dashes to separate levels in library names. Better support for prefixes.
 - Support haploid/hemizygous data
 - DNG-Call: Change default genotype likelihood parameters

## Library
 - Standardize as much as possible the interfaces for AD, BAM, and BCF. This requires changes to pileup, library, contig, stats, etc. sections.
 - Remove readgroups
 - Create LogProbability class
 - Change FindMutations to CallMutations
 - Change how AD stores values
 - Better interface for dealing with AlleleDepths and BCF files
 - Split the stats logic into several functions that we can reuse based on input
 - Add ploidy, library_label, and type to vertices in dng::detail::graph.
 - Distinguish maternal and paternal edge types
 - dng::rangeio helper namespace for sending a range to a stream
 - change how error exceptions are implemented

## Testing
 - Disable unit tests that no longer compile.
 - Update testdata sha
 - Improve the output of tests
 - Update the expected output of integration tests
 - Change unit test binary name and default options
 - Add unit tests for new classes
 - Rearrange unit-test specific headers
 - dng/likelihood.h -> genotyper.h
 - Wrapper for BCF synced reader
 - Create a dng::io::Ped class and change the way pedigrees are parsed, shifting some of the logic to RelationshipGraph
 - Add dng::io::File class
 - Additional utility classes
 - Keep track of extension in at_slurp
 - Rename some matrix classes
 - Expand mutation matrix creation functions to support diploid and haploid individuals
 - Support haploid genotype likelihoods
 - Change class names to clean up how regions are described and used
 - Update peeling to reflect different ploidies

## Miscellaneous

 - Add sublime project file to source code
